packages/backend/src/modules/user/user.service.ts

```diff
@@ -70,4 +83,107 @@ export default class UserService {
       _id: { $ne: exceptUser },
     });
   }
+
+  public async resetPasswordRequest(email: string): Promise<boolean> {
+    const user = await this.findByEmail(email);
+    if (!user) {
+      throw new UserInputError(`User with this email does not exist`);
+    }
+
+    const codeEntity = await this.createNewCodeAndDeletePrevious(email, CodeTypes.RESET_PASSWORD);
+
+    await this.sendgrid.send(
+      {
+        from: config.application.sendgrid.senderEmail,
+        to: email,
+        templateId: config.application.sendgrid.resetPasswordTemplateId,
+        dynamicTemplateData: {
+          resetPasswordUrl: `${config.client.host}/reset-password?code=${codeEntity.code}&email=${email}`,
+          firstName: user.firstName || user.email,
+        }
+      }
+    );
+
+    return true;
+  }
+
+  public async resetPassword(email: string, code: string, password: string): Promise<boolean> {
+    const isConfirmed = await this.acceptConfirmationCode(email, CodeTypes.RESET_PASSWORD, code);
+    if (!isConfirmed) throw new UserInputError('Confirmation code was not found');
+
+    const user = await this.findByEmail(email);
+
+    if (!user) {
+      throw new UserInputError(`User with email: ${email} was found`);
```

Comment:
Same as above, replace the message with "User with such email/password does not exist".